cmd/test-cli/main.go

```diff
@@ -186,28 +206,35 @@ func main() {
 		doGenerateValidator(validatorDataFile, gasLimit, validatorFeeRecipient)
 	case "register":
 		registerCommand.Parse(os.Args[2:])
-		doRegisterValidator(mustLoadValidator(validatorDataFile), boostEndpoint)
+		builderSigningDomain := computeDomain(boostTypes.DomainTypeAppBuilder, genesisForkVersionStr, boostTypes.Root{}.String())
+		doRegisterValidator(mustLoadValidator(validatorDataFile), boostEndpoint, builderSigningDomain)
 	case "getHeader":
 		getHeaderCommand.Parse(os.Args[2:])
-		doGetHeader(mustLoadValidator(validatorDataFile), boostEndpoint, createBeacon(isMergemock, beaconEndpoint, engineEndpoint), engineEndpoint)
+		builderSigningDomain := computeDomain(boostTypes.DomainTypeAppBuilder, genesisForkVersionStr, boostTypes.Root{}.String())
+		doGetHeader(mustLoadValidator(validatorDataFile), boostEndpoint, createBeacon(isMergemock, beaconEndpoint, engineEndpoint), engineEndpoint, builderSigningDomain)
 	case "getPayload":
 		getPayloadCommand.Parse(os.Args[2:])
-		genesisValidatorsRoot := boostTypes.Root(common.HexToHash(genesisValidatorsRootStr))
-		forkVersionBytes, err := hexutil.Decode(forkVersionStr)
-		if err != nil || len(forkVersionBytes) > 4 {
-			fmt.Println("Invalid fork version passed")
-			os.Exit(1)
-		}
-		var forkVersion [4]byte
-		copy(forkVersion[:], forkVersionBytes[:4])
-		signingDomain := boostTypes.ComputeDomain(boostTypes.DomainTypeBeaconProposer, forkVersion, genesisValidatorsRoot)
-		doGetPayload(mustLoadValidator(validatorDataFile), boostEndpoint, createBeacon(isMergemock, beaconEndpoint, engineEndpoint), engineEndpoint, signingDomain)
+		builderSigningDomain := computeDomain(boostTypes.DomainTypeAppBuilder, genesisForkVersionStr, boostTypes.Root{}.String())
+		proposerSigningDomain := computeDomain(boostTypes.DomainTypeBeaconProposer, bellatrixForkVersionStr, genesisValidatorsRootStr)
+		doGetPayload(mustLoadValidator(validatorDataFile), boostEndpoint, createBeacon(isMergemock, beaconEndpoint, engineEndpoint), engineEndpoint, builderSigningDomain, proposerSigningDomain)
 	default:
 		fmt.Println("Expected generate|register|getHeader|getPayload subcommand")
 		os.Exit(1)
 	}
 }
 
+func computeDomain(domainType boostTypes.DomainType, forkVersionHex string, genesisValidatorsRootHex string) boostTypes.Domain {
```

Comment:
should this function live in `go-builder-utils`? probably not important but just wondering

Reply:
Yeah, I don't think so since it's just a wrapper on internal options for usage with `boostTypes.ComputeDomain()`